sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -70,61 +70,98 @@ case class EnsureRequirements(
     val childrenIndexes = requiredChildDistributions.zipWithIndex.filter {
       case (UnspecifiedDistribution, _) => false
       case (_: BroadcastDistribution, _) => false
+      case (AllTuples, _) => false
```

Comment:
shall we also exclude `OrderedDistribution`? It does not have a real impact as in Spark there is no operator requires multiple `OrderedDistribution` today, but it makes the code cleaner.

Reply:
```
// Get the indexes of children which have specified distribution requirements that need to be
// co-partitioned.
val childrenIndexes = requiredChildDistributions.zipWithIndex.filter {
  case (_: ClusteredDistributuin, _) => true
  case _ => false
}
```

Reply:
Good idea!